src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -110,15 +120,31 @@ public CommandResult redo(Model model) throws CommandException {
      * Creates and returns a {@code Person} with the details of {@code personToEdit}
      * edited with {@code editPersonDescriptor}.
      */
-    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor) {
+    private static Person createEditedPerson(Person personToEdit, EditPersonDescriptor editPersonDescriptor)
+            throws CommandException {
         assert personToEdit != null;
 
         Name updatedName = editPersonDescriptor.getName().orElse(personToEdit.getName());
         Phone updatedPhone = editPersonDescriptor.getPhone().orElse(personToEdit.getPhone());
         Email updatedEmail = editPersonDescriptor.getEmail().orElse(personToEdit.getEmail());
         Category updatedCategory = editPersonDescriptor.getCategory().orElse(personToEdit.getCategory());
+        Person editedPerson = PersonFactory.createPerson(updatedName, updatedPhone, updatedEmail, updatedCategory);
+
+        if (editedPerson instanceof Staff) {
+            Staff editedStaff = (Staff) editedPerson;
+            Group updatedGroup = editPersonDescriptor.getGroup().orElse(new Group(editedStaff.getGroupNumber()));
+            editedStaff.setGroupNumber(updatedGroup.getGroupNumber());
+            return editedStaff;
+        } else if (editedPerson instanceof Participant) {
+            Participant editedParticipant = (Participant) editedPerson;
+            Group updatedGroup = editPersonDescriptor.getGroup().orElse(new Group(editedParticipant.getGroupNumber()));
+            editedParticipant.setGroupNumber(updatedGroup.getGroupNumber());
+            return editedParticipant;
```

Comment:
I think this can be done better by moving these code into Person and subclasses.